### ISSUE TITLE
Can't change LBaaS properties, incorrect re-creation plan generated when using ignore_changes

### ISSUE THREAD

**Issue by stanicek** [terraform-providers/terraform-provider-oci#666]
### Terraform Version
Terraform v0.11.8

### OCI Provider Version
+ provider.oci v3.8.0

### Description:
Given: Load balancer up and running. 
```
resource "oci_load_balancer_certificate" "certificate" {
  load_balancer_id   = "${oci_load_balancer_load_balancer.my_load_balancer.id}"
  certificate_name   = "cert"
  public_certificate = "${file("domain.crt")}"
  private_key        = "${file("domain.key")}"
  ca_certificate     = "${file("ca.crt")}"
}

resource "oci_load_balancer_listener" "listener_ssl" {
  load_balancer_id         = "${oci_load_balancer_load_balancer.my_load_balancer.id}"
  name                     = "mylistener"
  protocol                 = "HTTP"
  port                     = "444"
  default_backend_set_name = "${oci_load_balancer_backend_set.backend_set_ssl.name}"

  ssl_configuration {
    certificate_name        = "${oci_load_balancer_certificate.certificate.certificate_name}"
    verify_peer_certificate = false
  }

  lifecycle {
    ignore_changes = ["ssl_configuration.0.certificate_name"]
  }
}

resource "oci_load_balancer_backend_set" "backend_set_ssl" {
  load_balancer_id = "${oci_load_balancer_load_balancer.my_load_balancer.id}"
  name             = "mybackendset"
  policy           = "ROUND_ROBIN"

  health_checker = {
    protocol          = "HTTP"
    port              = "80"
    interval_ms       = "30000"
    timeout_in_millis = "3000"
    url_path          = "/"
  }

  ssl_configuration {
    certificate_name        = "${oci_load_balancer_certificate.certificate.certificate_name}"
    verify_peer_certificate = false
  }

  lifecycle {
    ignore_changes = ["ssl_configuration.0.certificate_name"]
  }
}
```

When: I want to change a load balancer property which triggers load balancer re-creation (e.g. shape, subnet, etc.) 

Actual: Terraform plans change only of load balancer and certificate resource. 
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ oci_load_balancer_certificate.certificate (new resource required)
      id:                   "cert" => <computed> (forces new resource)
      ca_certificate:       "..."
      certificate_name:     "cert" => "cert"
      load_balancer_id:     "ocid1.loadbalancer...." => "${oci_load_balancer_load_balancer.my_load_balancer.id}" (forces new resource)
      private_key:          <sensitive> => <sensitive> (attribute changed)
      public_certificate:   "..."
      state:                "SUCCEEDED" => <computed>

-/+ oci_load_balancer_load_balancer.my_load_balancer (new resource required)
      id:                   "ocid1.loadbalancer...." => <computed> (forces new resource)
      compartment_id:       "ocid1.compartment...." => "ocid1.compartment...."
      display_name:         "testlb" => "testlb"
      freeform_tags.%:      "0" => <computed>
      ip_address_details.#: "1" => <computed>
      ip_addresses.#:       "1" => <computed>
      is_private:           "true" => "true"
      shape:                "100Mbps" => "400Mbps" (forces new resource)
      state:                "ACTIVE" => <computed>
      subnet_ids.#:         "1" => "1"
      subnet_ids.0:         "ocid1.subnet...." => "ocid1.subnet...."
      time_created:         "2018-12-07 03:17:48.371 +0000 UTC" => <computed>


Plan: 2 to add, 0 to change, 2 to destroy.
```
This ends up with error:
```
1 error(s) occurred:
* oci_load_balancer_certificate.certificate (destroy): 1 error(s) occurred:
* oci_load_balancer_certificate.certificate: WorkRequest FAILED: [{ ErrorCode=BAD_INPUT Message=Listener for 'mylistener' refers to unknown certificate 'cert' }]
```
Expected: Terraform plans change of load balancer, certificate, listener and backend set resource, so everything is recreated properly.

Workaround is to temporarily comment out life cycle sections.

**Comment by ravitandonrt** [comment]
@stanicek Can you comment more on the use case for ignoring lifecycle changes?

What you want to do instead is to set `create_before_destroy` if you want to rotate certificate on the load balancer. The error you mention seems to indicate it is due to the association of the certificate to the load balancer.

Doc link: https://www.terraform.io/docs/providers/oci/r/load_balancer_certificate.html
Example link: https://github.com/terraform-providers/terraform-provider-oci/blob/master/docs/examples/load_balancer/lb_full/lb_full.tf#L242

```
oci_load_balancer_certificate
This resource provides the Certificate resource in Oracle Cloud Infrastructure Load Balancer service.

Creates an asynchronous request to add an SSL certificate bundle.

Set the terraform flag lifecycle { create_before_destroy = true } in your certificate to facilitate rotating certificates. A certificate cannot be deleted if it is attached to another resource (a listener or a backend set for example). Because certificate_name in the listener is an updatable parameter, terraform will attempt to recreate the certificate first and then update the listener but the certificate cannot be deleted while it is attached to a listener so it will fail. Setting the flag makes it so that when a certificate is recreated, the new certificate will be created first before the old one gets deleted. Whenever you change any values on a certificate that causes it to be recreated the certificate_name MUST also change. Otherwise you will get an error saying that a certificate with that name already exists.
```

**Comment by afedorch** [comment]
@stanicek we are keeping the issue open to verify that there are no problems with the new terraform release for this case.
see https://github.com/hashicorp/terraform/issues/19670

more convenient workaround for this issue could be using the `taint` command: 
https://www.terraform.io/docs/commands/taint.html

these commands should allow to recreate all the resources:

```
terraform taint oci_load_balancer_backend_set.backend_set_ssl
terraform taint oci_load_balancer_listener.listener_ssl
terrafrom apply
```

**Comment by afedorch** [comment]
The reported example works for the same provider version with the v0.12 alpha binaries from https://releases.hashicorp.com/terraform/0.12.0-alpha4/ 
(small changes in the .tf syntax are required, for an example `ssl_configuration.0.certificate_name` -> `ssl_configuration.certificate_name` to use terraform v0.12)

@stanicek you should be able to use `ignore_changes` with the released terraform v0.12 (schedule is still TBD).
